From 5db492cd84f30b41c0ae8c107fc5b5ed61c3cea9 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 12 Oct 2015 03:32:51 -0400
Subject: [PATCH] ipv6.c: replace 'inline' with 'static inline' for gcc 5.x

gcc 5.x defaults to -std=gnu11 instead of -std=gnu89 which causes
semantics for inline functions changes.

The standalone 'inline' causes error with gcc 5 such as:

git/testcases/kernel/syscalls/kill/kill10.c:355: undefined reference to `k_sigaction'

Replace inline with static inline to be compatible with both gcc 4 and 5.

Upstream-Status: Backport [From https://github.com/linux-test-project/ltp/commit/40a2457cb8ec42a05a2f96b0810057efdb2a55f5]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 iscsiuio/src/uip/ipv6.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/iscsiuio/src/uip/ipv6.c b/iscsiuio/src/uip/ipv6.c
index 594495a..75db4ba 100644
--- a/iscsiuio/src/uip/ipv6.c
+++ b/iscsiuio/src/uip/ipv6.c
@@ -46,7 +46,7 @@
 #include "uipopt.h"
 #include "dhcpv6.h"
 
-inline int best_match_bufcmp(u8_t * a, u8_t * b, int len)
+static inline int best_match_bufcmp(u8_t * a, u8_t * b, int len)
 {
 	int i;
 
-- 
1.9.1

